src/main/java/gregtech/api/recipes/logic/OverclockingLogic.java

```diff
@@ -58,27 +58,28 @@ public static int[] standardOverclockingLogic(int recipeEUt, long maxVoltage, in
      * @param requiredTemp the required temperature of the recipe
      * @return the amount of EU/t discounts to apply
      */
-    public static int calculateAmountCoilEUtDiscount(int providedTemp, int requiredTemp) {
+    private static int calculateAmountCoilEUtDiscount(int providedTemp, int requiredTemp) {
         return Math.max(0, (providedTemp - requiredTemp) / COIL_EUT_DISCOUNT_TEMPERATURE);
     }
 
     /**
+     * Handles applying the coil EU/t discount. Call before overclocking.
+     *
      * @param recipeEUt the EU/t of the recipe
-     * @param amountEUtDiscount the amount of discounts to apply
+     * @param providedTemp the temperate provided by the machine
+     * @param requiredTemp the required temperature of the recipe
      * @return the discounted EU/t
      */
-    public static int applyCoilEUtDiscount(int recipeEUt, int amountEUtDiscount) {
+    public static int applyCoilEUtDiscount(int recipeEUt, int providedTemp, int requiredTemp) {
+        if (requiredTemp < OverclockingLogic.COIL_EUT_DISCOUNT_TEMPERATURE) return recipeEUt;
+        int amountEUtDiscount = OverclockingLogic.calculateAmountCoilEUtDiscount(providedTemp, requiredTemp);
         if (amountEUtDiscount < 1) return recipeEUt;
```

Comment:
This line is now unneeded, as you check the required temp earlier now.

Reply:
This line is technically unneeded to achieve the same result, but is still useful as an early exit. 

For example, in a scenario such as providedTemp=901, and requiredTemp=900: 
The first if statement would be false, since `900 < COIL_EUT_DISCOUNT_TEMPERATURE = 900 < 900 = false`.

Then call `calculateAmountCoilEUtDiscount()` would be called. That method would return 0 in this case, since `(901 - 900) / 900 = 0`.

Then if statement is checked to skip the math at the end, since any number to the 0th power is 1, which would not modify the EUt.